src/plugins/intel_gpu/src/graph/graph_optimizer/add_required_reorders.cpp

```diff
@@ -246,18 +246,42 @@ void add_required_reorders::run(program& p) {
         }
 
         // Remove padded-inputs in spatial axes not to use ref kernel which causes huge perf drop
-        if (usr->is_type<mvn>() && usr->as<mvn>().input().is_padded_spatial()) {
-            auto out_layout = usr->get_output_layout();
-            // Check formats of implemented opt kernels without a spatial padding support
-            if (out_layout.format == format::b_fs_yx_fsv16 || out_layout.format == format::b_fs_zyx_fsv16 ||
-                out_layout.format == format::bs_fs_yx_bsv32_fsv16 || out_layout.format == format::bs_fs_yx_bsv32_fsv32) {
-                auto& dep = usr->as<mvn>().input();
-                cldnn::layout layout_wo_padding = dep.get_output_layout();
-                layout_wo_padding.data_padding = cldnn::padding{};
-                auto new_reorder = std::make_shared<reorder>(dep.id() + "_no_pad_reorder", dep.id(), layout_wo_padding);
-                auto& new_reorder_node = p.get_or_create(new_reorder);
-                p.add_intermediate(new_reorder_node, *usr, dep);
-                new_reorder_node.recalc_output_layout(false);
+        if (usr->is_type<mvn>()) {
+            if (usr->as<mvn>().input().is_padded_spatial()) {
+                auto out_layout = usr->get_output_layout();
+                // Check formats of implemented opt kernels without a spatial padding support
+                if (out_layout.format == format::b_fs_yx_fsv16 || out_layout.format == format::b_fs_zyx_fsv16 ||
+                    out_layout.format == format::bs_fs_yx_bsv32_fsv16 || out_layout.format == format::bs_fs_yx_bsv32_fsv32) {
+                    auto& dep = usr->as<mvn>().input();
+                    cldnn::layout layout_wo_padding = dep.get_output_layout();
+                    layout_wo_padding.data_padding = cldnn::padding{};
+                    auto new_reorder = std::make_shared<reorder>(dep.id() + "_no_pad_reorder", dep.id(), layout_wo_padding);
+                    auto& new_reorder_node = p.get_or_create(new_reorder);
+                    p.add_intermediate(new_reorder_node, *usr, dep);
+                    new_reorder_node.recalc_output_layout(false);
+                }
+            }
+
+            auto input_layout = usr->get_input_layout();
+            auto input_pshape = input_layout.get_partial_shape();
+            auto prim = usr->as<mvn>().get_primitive();
+
+            if (prim->requires_alignment(input_pshape)) {
+                auto block_sizes = format::block_sizes(input_layout.format);
+                auto axes = prim->reduction_axes;
+                if (input_layout.is_dynamic() || block_sizes.size() > 1
+                    || (block_sizes.size() == 1 &&
+                        input_pshape[block_sizes[0].first].get_length() % block_sizes[0].second != 0 &&
+                        (axes.size() == 1 && static_cast<size_t>(axes[0]) != block_sizes[0].first))) {
+                    auto rank = input_pshape.size();
```

Comment:
plz change the condition to “if block axis is not included in axes”

Reply:
flattening only supports single blocked format. right? Can reduction axes have multiple values when the format is single blocked?

Reply:
Changed.